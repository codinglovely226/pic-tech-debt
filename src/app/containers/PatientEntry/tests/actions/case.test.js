import {
  replacePatientMedicalRecordNumber,
  replacePatientName,
  updatePatientAge,
  updatePatientAgeGroup,
  updatePatientAgeYears,
  updatePatientAnimalType,
  updatePatientCategory,
  updatePatientCircumstance,
  updatePatientLocation,
  updatePatientMedicalRecordNumber,
  updatePatientName,
  updatePatientOccupation,
  updatePatientPoisonSeverityScore,
  updatePatientPlaceId,
  updatePatientAddress1,
  updatePatientAddress2,
  updatePatientSuburb,
  updatePatientState,
  updatePatientPostcode,
  updatePatientCountry,
  updatePatientISOCountry,
  updatePatientTelephone,
  updatePatientGeometry,
  updatePatientPregnantLactating,
  updatePatientPregnantStatus,
  updatePatientPregnantTrimester,
  updatePatientPregnantTrimesterUnknown,
  updatePatientRiskAssessment,
  updatePatientSex,
  updatePatientWeight,
} from '../../actions/case';

describe('Patient action creators', () => {
  const appKey = { id: 'appKeyId', key: 'appKey' };
  it('replacePatientMedicalRecordNumber', () => {
    expect(replacePatientMedicalRecordNumber(1, 'mrn', appKey)).toMatchSnapshot();
  });
  it('replacePatientName', () => {
    expect(replacePatientName(1, 'Patient Name', appKey)).toMatchSnapshot();
  });
  it('updatePatientAge', () => {
    expect(updatePatientAge(1, 10, 20, 'months', 'years')).toMatchSnapshot();
  });
  it('updatePatientAgeGroup', () => {
    expect(updatePatientAgeGroup(1, 'months', 'years')).toMatchSnapshot();
  });
  it('updatePatientAgeYears', () => {
    expect(updatePatientAgeYears(1, '', '0.75')).toMatchSnapshot();
  });
  it('updatePatientAnimalType', () => {
    expect(updatePatientAnimalType(1, 'dog', 'bird')).toMatchSnapshot();
  });
  it('updatePatientCategory', () => {
    expect(updatePatientCategory(1, 'animal', 'human')).toMatchSnapshot();
  });
  it('updatePatientCircumstance', () => {
    expect(updatePatientCircumstance(1, 'Old Circumstance', 'New Circumstance')).toMatchSnapshot();
  });
  it('updatePatientLocation', () => {
    expect(updatePatientLocation(1, 'Old Location', 'New Location')).toMatchSnapshot();
  });
  it('updatePatientMedicalRecordNumber', () => {
    expect(updatePatientMedicalRecordNumber(1, '1234', '5678')).toMatchSnapshot();
  });
  it('updatePatientName', () => {
    expect(updatePatientName(1, 'Old Name', 'New Name')).toMatchSnapshot();
  });
  it('updatePatientOccupation', () => {
    expect(updatePatientOccupation(1, 'Old Occupation', 'New Occupation')).toMatchSnapshot();
  });
  it('updatePatientPoisonSeverityScore', () => {
    expect(updatePatientPoisonSeverityScore(1, 'Minimal', 'Minor')).toMatchSnapshot();
  });
  it('updatePatientPlaceId', () => {
    expect(updatePatientPlaceId(1, 'Old Place Id', 'New Place Id')).toMatchSnapshot();
  });
  it('updatePatientAddress1', () => {
    expect(updatePatientAddress1(1, 'Old Address1', 'New Address 1')).toMatchSnapshot();
  });
  it('updatePatientAddress2', () => {
    expect(updatePatientAddress2(1, 'Old Address1', 'New Address 1')).toMatchSnapshot();
  });
  it('updatePatientSuburb', () => {
    expect(updatePatientSuburb(1, 'Perth', 'Sydney')).toMatchSnapshot();
  });
  it('updatePatientState', () => {
    expect(updatePatientState(1, 'WA', 'NSW')).toMatchSnapshot();
  });
  it('updatePatientPostcode', () => {
    expect(updatePatientPostcode(1, '6000', '3000')).toMatchSnapshot();
  });
  it('updatePatientCountry', () => {
    expect(updatePatientCountry(1, 'Australia', 'New Zealand')).toMatchSnapshot();
  });
  it('updatePatientISOCountry', () => {
    expect(updatePatientISOCountry(1, 'AU', 'NZ')).toMatchSnapshot();
  });
  it('updatePatientTelephone', () => {
    expect(updatePatientTelephone(1, '123456', '56789')).toMatchSnapshot();
  });
  it('updatePatientGeometry', () => {
    expect(updatePatientGeometry(1, {}, {})).toMatchSnapshot();
  });
  it('updatePatientPregnantLactating', () => {
    expect(updatePatientPregnantLactating(1, 'No', 'Yes')).toMatchSnapshot();
  });
  it('updatePatientPregnantStatus', () => {
    expect(updatePatientPregnantStatus(1, 'No', 'Yes')).toMatchSnapshot();
  });
  it('updatePatientPregnantTrimester', () => {
    expect(updatePatientPregnantTrimester(1, '1', '2')).toMatchSnapshot();
  });
  it('updatePatientPregnantTrimesterUnknown', () => {
    expect(updatePatientPregnantTrimesterUnknown(1, false, true)).toMatchSnapshot();
  });
  it('updatePatientRiskAssessment', () => {
    expect(updatePatientRiskAssessment(1, 'Old Assessment', 'New Assessment')).toMatchSnapshot();
  });
  it('updatePatientSex', () => {
    expect(updatePatientSex(1, 'male', 'female')).toMatchSnapshot();
  });
  it('updatePatientWeight', () => {
    expect(updatePatientWeight(1, '80', '83', 'kg', 'kg')).toMatchSnapshot();
  });
});
