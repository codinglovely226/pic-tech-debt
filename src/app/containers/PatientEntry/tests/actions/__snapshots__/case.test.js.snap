exports[`Patient action creators replacePatientMedicalRecordNumber 1`] = `
Object {
  "appKey": Object {
    "id": "appKeyId",
    "key": "appKey",
  },
  "id": 1,
  "medicalRecordNumber": "mrn",
  "type": "pic/PatientEntry/REPLACE_PATIENT_MEDICAL_RECORD_NUMBER",
}
`;

exports[`Patient action creators replacePatientName 1`] = `
Object {
  "appKey": Object {
    "id": "appKeyId",
    "key": "appKey",
  },
  "id": 1,
  "name": "Patient Name",
  "type": "pic/PatientEntry/REPLACE_PATIENT_NAME",
}
`;

exports[`Patient action creators updatePatientAddress1 1`] = `
Object {
  "address1": "New Address 1",
  "id": 1,
  "old": "Old Address1",
  "type": "pic/PatientEntry/UPDATE_PATIENT_ADDRESS1",
}
`;

exports[`Patient action creators updatePatientAddress2 1`] = `
Object {
  "address2": "New Address 1",
  "id": 1,
  "old": "Old Address1",
  "type": "pic/PatientEntry/UPDATE_PATIENT_ADDRESS2",
}
`;

exports[`Patient action creators updatePatientAge 1`] = `
Object {
  "age": 20,
  "ageUnits": "years",
  "id": 1,
  "oldAge": 10,
  "oldAgeUnits": "months",
  "type": "pic/PatientEntry/UPDATE_PATIENT_AGE",
}
`;

exports[`Patient action creators updatePatientAgeGroup 1`] = `
Object {
  "ageGroup": "years",
  "id": 1,
  "old": "months",
  "type": "pic/PatientEntry/UPDATE_PATIENT_AGE_GROUP",
}
`;

exports[`Patient action creators updatePatientAgeYears 1`] = `
Object {
  "ageYears": "0.75",
  "id": 1,
  "old": "",
  "type": "pic/PatientEntry/UPDATE_PATIENT_AGE_YEARS",
}
`;

exports[`Patient action creators updatePatientAnimalType 1`] = `
Object {
  "animalType": "bird",
  "id": 1,
  "old": "dog",
  "type": "pic/PatientEntry/UPDATE_PATIENT_ANIMAL_TYPE",
}
`;

exports[`Patient action creators updatePatientCategory 1`] = `
Object {
  "category": "human",
  "id": 1,
  "old": "animal",
  "type": "pic/PatientEntry/UPDATE_PATIENT_CATEGORY",
}
`;

exports[`Patient action creators updatePatientCircumstance 1`] = `
Object {
  "circumstance": "New Circumstance",
  "id": 1,
  "old": "Old Circumstance",
  "type": "pic/PatientEntry/UPDATE_PATIENT_CIRCUMSTANCE",
}
`;

exports[`Patient action creators updatePatientCountry 1`] = `
Object {
  "country": "New Zealand",
  "id": 1,
  "old": "Australia",
  "type": "pic/PatientEntry/UPDATE_PATIENT_COUNTRY",
}
`;

exports[`Patient action creators updatePatientGeometry 1`] = `
Object {
  "geometry": Object {},
  "id": 1,
  "old": Object {},
  "type": "pic/PatientEntry/UPDATE_PATIENT_GEOMETRY",
}
`;

exports[`Patient action creators updatePatientISOCountry 1`] = `
Object {
  "id": 1,
  "isoCountry": "NZ",
  "old": "AU",
  "type": "pic/PatientEntry/UPDATE_PATIENT_ISOCOUNTRY",
}
`;

exports[`Patient action creators updatePatientLocation 1`] = `
Object {
  "id": 1,
  "location": "New Location",
  "old": "Old Location",
  "type": "pic/PatientEntry/UPDATE_PATIENT_LOCATION",
}
`;

exports[`Patient action creators updatePatientMedicalRecordNumber 1`] = `
Object {
  "appKey": undefined,
  "id": 1,
  "medicalRecordNumber": "5678",
  "old": "1234",
  "type": "pic/PatientEntry/UPDATE_PATIENT_MEDICAL_RECORD_NUMBER",
}
`;

exports[`Patient action creators updatePatientName 1`] = `
Object {
  "appKey": undefined,
  "id": 1,
  "name": "New Name",
  "old": "Old Name",
  "type": "pic/PatientEntry/UPDATE_PATIENT_NAME",
}
`;

exports[`Patient action creators updatePatientOccupation 1`] = `
Object {
  "id": 1,
  "occupation": "New Occupation",
  "old": "Old Occupation",
  "type": "pic/PatientEntry/UPDATE_PATIENT_OCCUPATION",
}
`;

exports[`Patient action creators updatePatientPlaceId 1`] = `
Object {
  "id": 1,
  "old": "Old Place Id",
  "placeId": "New Place Id",
  "type": "pic/PatientEntry/UPDATE_PATIENT_PLACEID",
}
`;

exports[`Patient action creators updatePatientPoisonSeverityScore 1`] = `
Object {
  "id": 1,
  "old": "Minimal",
  "poisonSeverityScore": "Minor",
  "type": "pic/PatientEntry/UPDATE_PATIENT_POISON_SEVERITY_SCORE",
}
`;

exports[`Patient action creators updatePatientPostcode 1`] = `
Object {
  "id": 1,
  "old": "6000",
  "postcode": "3000",
  "type": "pic/PatientEntry/UPDATE_PATIENT_POSTCODE",
}
`;

exports[`Patient action creators updatePatientPregnantLactating 1`] = `
Object {
  "id": 1,
  "old": "No",
  "pregnantLactating": "Yes",
  "type": "pic/PatientEntry/UPDATE_PATIENT_PREGNANT_LACTATING",
}
`;

exports[`Patient action creators updatePatientPregnantStatus 1`] = `
Object {
  "id": 1,
  "old": "No",
  "pregnantStatus": "Yes",
  "type": "pic/PatientEntry/UPDATE_PATIENT_PREGNANT_STATUS",
}
`;

exports[`Patient action creators updatePatientPregnantTrimester 1`] = `
Object {
  "id": 1,
  "old": "1",
  "pregnantTrimester": "2",
  "type": "pic/PatientEntry/UPDATE_PATIENT_PREGNANT_TRIMESTER",
}
`;

exports[`Patient action creators updatePatientPregnantTrimesterUnknown 1`] = `
Object {
  "id": 1,
  "old": false,
  "pregnantTrimesterUnknown": true,
  "type": "pic/PatientEntry/UPDATE_PATIENT_PREGNANT_TRIMESTER_UNKNOWN",
}
`;

exports[`Patient action creators updatePatientRiskAssessment 1`] = `
Object {
  "id": 1,
  "old": "Old Assessment",
  "riskAssessment": "New Assessment",
  "type": "pic/PatientEntry/UPDATE_PATIENT_RISK_ASSESSMENT",
}
`;

exports[`Patient action creators updatePatientSex 1`] = `
Object {
  "gender": "female",
  "id": 1,
  "old": "male",
  "type": "pic/PatientEntry/UPDATE_PATIENT_SEX",
}
`;

exports[`Patient action creators updatePatientState 1`] = `
Object {
  "id": 1,
  "old": "WA",
  "state": "NSW",
  "type": "pic/PatientEntry/UPDATE_PATIENT_STATE",
}
`;

exports[`Patient action creators updatePatientSuburb 1`] = `
Object {
  "id": 1,
  "old": "Perth",
  "suburb": "Sydney",
  "type": "pic/PatientEntry/UPDATE_PATIENT_SUBURB",
}
`;

exports[`Patient action creators updatePatientTelephone 1`] = `
Object {
  "id": 1,
  "old": "123456",
  "telephone": "56789",
  "type": "pic/PatientEntry/UPDATE_PATIENT_TELEPHONE",
}
`;

exports[`Patient action creators updatePatientWeight 1`] = `
Object {
  "id": 1,
  "oldWeight": "80",
  "oldWeightUnits": "kg",
  "type": "pic/PatientEntry/UPDATE_PATIENT_WEIGHT",
  "weight": "83",
  "weightUnits": "kg",
}
`;
